src/deferred.js

```diff
@@ -13,6 +13,23 @@ function Thrower( ex ) {
 	throw ex;
 }
 
+function adoptValue( value, resolve, reject ) {
+	var method;
+
+	// Check for promise aspect first to privilege synchronous behavior
+	if ( value && jQuery.isFunction( ( method = value.promise ) ) ) {
```

Comment:
>  or just let it be, but the line MUST take that form if we guarantee proper 

I think such changes should be supported and confirmed by the tests, otherwise all of it could be seen as speculation or subjective interpretation, it seems this discussion revealed luck of it, so i'd say either in this pull or right after this one, we need to add those tests for `jQuery.when`.